src/client.cpp

```diff
@@ -219,9 +221,15 @@ namespace quicr {
                                         "Received unsubscribe conn_id: {0} subscribe_id: {1}",
                                         conn_ctx.connection_handle,
                                         msg.subscribe_id);
-                    sub_it->second.get()->SetStatus(SubscribeTrackHandler::Status::kNotSubscribed);
 
-                    RemoveSubscribeTrack(conn_ctx, *sub_it->second);
+                    const auto pub_track_ns_it =
```

Comment:
This is pretty ugly. While this works, I'm sure there is a much cleaner way to do the same thing.